cmd/client-keystone-auth/main.go

```diff
@@ -54,53 +55,64 @@ const respTemplate string = `{
 }`
 
 func promptForString(field string, r io.Reader, show bool) (result string, err error) {
-	fmt.Printf("Please enter %s: ", field)
+	// We have to print output to Stderr, because Stdout is redirected and not shown to the user.
+	fmt.Fprintf(os.Stderr, "Please enter %s: ", field)
```

Comment:
Using 'os.Stderr'? Is there other better way to do it?

Reply:
Good question... We can't use stdout anyway, so we have to invent some sort of hack to show this output to the user.
I'm not proud that we have to do this trick, but I don't see any better solution right now... Leaving the user without any prompt strings is very annoying.

Reply:
Forgoing the "spirit of stderr and stdout" plenty of other auth tools do this

```
$ gcloud auth login > /dev/null
Your browser has been opened to visit:

    https://accounts.google.com/o/oauth2/auth?redirect_uri=(stuff)


        WARNING: `gcloud auth login` no longer writes application default credentials.
If you need to use ADC, see:
  gcloud auth application-default --help

You are now logged in as [eric.chiang.m@gmail.com].
Your current project is [None].  You can change this setting by running:
  $ gcloud config set project PROJECT_ID
```

I also find it annoying when tools prompt me over stdout. It makes it impossible to do something like:

```
tool-that-requires-passwd | xclip -selection c
```

Your prompt ends up piped into the next tool, which doesn't seem like the intention.

My vote is: leave stdout for things you'd actually want to pipe into the next command. Use stderr from prompting.

Reply:
I agree with @ericchiang ! This is just what this code does - uses stderr instead of stdout to print user-facing messages.

I hope @dims and @FengyunPan2 don't have any objections.